lib/triton-tags.js

```diff
@@ -0,0 +1,185 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright 2016, Joyent, Inc.
+ */
+
+/*
+ * In Triton some per-VM configuration is controlled via special structured
+ * tags on the VM's "tags" object. They are all prefixed with "triton.".
+ * Let's call them "Triton tags".
+ *
+ * Triton tags are all optional.
+ *
+ * - `triton.cns.disable` (boolean): Can be set on a VM to tell the CNS service
+ *   to not serve records for this VM.
+ * - `triton.cns.services` (string): Comma-separated list of DNS-name strings
+ *   for the CNS service.
+ * - `triton.cns.reverse_ptr` (string): DNS reverse pointer for this VM. Used
+ *   by the CNS service.
+ */
+
+var assert = require('assert-plus');
+var format = require('util').format;
+
+
+/*
+ * For now, using the more limited labels allowed by RFC1123. RFC2181 supercedes
+ * 1123, but the broader range of characters can sometimes cause problems with
+ * other systems (e.g. see the underscore in RFC5321).
+ */
+var DNS_NAME_RE = /^[a-z0-9][a-z0-9\-]{0,62}(?:\.[a-z0-9][a-z0-9\-]{0,62})*$/i;
+
+const TRITON_TAG_PREFIX = 'triton.';
+
+var typeFromKey = {
+    'triton.cns.services': 'string',
```

Comment:
That's fine. I'd suggest taking out the TRITON_TAG_PREFIX then and just using 'triton.' everywhere. So that someone doesn't think they can actually change this meaningfully in one place. Or add a comment saying that this is only a convenience thing and in order to change one would still need to replace all instances of `triton.` everywhere.